### PR TITLE
General: use client-side round-robin load balancing for grpc

### PR DESCRIPTION
**Pull request by BojanZelic**
Use Client-side grpc round-robin load balancing to better spread the load when using the external scaler;

**Scenario A:**
keda operator makes a grpc call to a Service with a clusterIP. There's only 1 clusterIP. This PR won't affect this setup. Load balancing (within k8s/iptables) only happens when new connections or a connection failure. Requests are hitting the same underlying pod; Users should migrate to headless service.

**Scenario B:**
Keda operator makes a grpc call to a headless service & DNS randomizes the pod IP order. load balancing happens whenever there's a connection failure or if the server set a MaxConnectionAge, otherwise 1 pod is going to be receiving the all of the load. This PR fixes this

**Scenario C:**
Keda operator makes a grpc call to a headless service & DNS doesn't randomize the pod IP order. load balancing currently never happens, 1 pod will be receiving all the load. This PR fixes this. 

### Checklist

- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes # https://github.com/kedacore/keda/issues/5224

### CODE REVIEW

File: pkg/metricsservice/client.go
Comment (JorTurFer) [inline pkg/metricsservice/client.go:41]: I'm not sure if here makes sense as this is used for calling from metrics server to operator and it'll have a single endpoint available because it depends on leader election API.

**Review by JorTurFer — COMMENTED** [review]
This LGTM but I have a really poor knowledge about gRPC
WDYT @zroubalik ?

File: pkg/metricsservice/client.go
Comment (BojanZelic-DD) [reply pkg/metricsservice/client.go]: thanks that's a good callout, removed since it's unnecessary

**Comment by BojanZelic-DD** [comment]
The E2E test failure seems unrelated? Seems like it failed on the cleanup steps

**Comment by JorTurFer** [comment]
> The E2E test failure seems unrelated? Seems like it failed on the cleanup steps

e2e tests are unstable right now, [I'm working on a PR](https://github.com/kedacore/keda/pull/5226) to fix it, once the fix is merged I'll trigger e2e test again here

**Review by zroubalik — APPROVED** [review]
LGTM

**Comment by zroubalik** [comment]
@BojanZelic please rebase and resolve conflicts, let's merge this if e2e pass then. Thanks!

**Pull request merged**
